client/gutenberg/extensions/markdown/editor.js

```diff
@@ -15,46 +15,50 @@ import save from './save';
 import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
 import JetpackBlockType from 'gutenberg/extensions/presets/jetpack/utils/jetpack-block-type';
 
-const MarkdownBlock = new JetpackBlockType( 'markdown', {
-	title: __( 'Markdown' ),
-
-	description: (
-		<Fragment>
-			<p>{ __( 'Use regular characters and punctuation to style text, links, and lists.' ) }</p>
-			<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
-				{ __( 'Support reference' ) }
-			</ExternalLink>
-		</Fragment>
-	),
-
-	icon: (
-		<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
-			<rect
-				width="198"
-				height="118"
-				x="5"
-				y="5"
-				ry="10"
-				stroke="currentColor"
-				strokeWidth="10"
-				fill="none"
-			/>
-			<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
-		</svg>
-	),
-
-	category: 'jetpack',
-
-	keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
-
-	attributes: {
-		//The Markdown source is saved in the block content comments delimiter
-		source: { type: 'string' },
-	},
+const MarkdownBlock = new JetpackBlockType( {
+	name: 'markdown',
+	requiredBlock: 'markdown',
+	config: {
```

Comment:
At `registerBlockType` this is called `settings`, should we align naming?

https://github.com/WordPress/gutenberg/blob/57a7405a89dbd6ad7763aa04dcd2aabdd26b25b9/packages/blocks/src/api/registration.js#L55-L66

Reply:
Or perhaps we should call it `blockTypeSettings` if in future we'd also have support for registering plugins via this method?

https://github.com/WordPress/gutenberg/blob/3ff9890fd753de5b3c3ea623ecf862ad252f11cb/packages/plugins/README.md

E.g. Publicize is a plugin, not a block type and also needs checks like this.

If we're planning to do that, perhaps we should rename `JetpackBlockType` to `JetpackExtension` or something like that instead?

Reply:
`registerBlockStyle` and `addFilter` might also need to be considered as future extension points here.